x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/common/EmbeddingRequestChunker.java

```diff
@@ -177,9 +215,31 @@ public void onResponse(InferenceServiceResults inferenceServiceResults) {
                 int start = 0;
                 for (var pos : positions) {
                     results.get(pos.inputIndex())
-                        .setOnce(pos.chunkIndex(), textEmbeddingResults.embeddings().subList(start, start + pos.embeddingCount()));
+                        .setOnce(pos.chunkIndex(), embeddingResults.embeddings().subList(start, start + pos.embeddingCount()));
                     start += pos.embeddingCount();
                 }
+
+                if (firstResultType.get() == null) {
+                    if (inferenceServiceResults instanceof TextEmbeddingResults) {
+                        firstResultType.set(EmbeddingType.FLOAT);
+                    } else if (inferenceServiceResults instanceof TextEmbeddingByteResults) {
+                        firstResultType.set(EmbeddingType.BYTE);
+                    } else if (inferenceServiceResults instanceof SparseEmbeddingResults) {
+                        firstResultType.set(EmbeddingType.SPARSE);
+                    }
```

Comment:
From the above comment, what happens if we don't know the type? Do we just ignore it?